girder_worker/cli/__main__.py

```diff
@@ -0,0 +1,94 @@
+import click
+import pkg_resources as pr
+from click.core import Command, Argument, Option
+from girder_worker.entrypoint import get_extensions, get_extension_tasks, import_all_includes
+from stevedore import driver
+from girder_worker_utils.decorators import (
+    GWFuncDesc,
+    Arg,
+    KWArg)
+
+GWRUN_ENTRYPOINT_GROUP = 'gwrun_output_handlers'
+
+def _cast_to_command(f):
+    if isinstance(f, Command):
+        return f
+
+    description = GWFuncDesc.get_description(f)
+    if description is not None:
+        for arg in reversed(description.arguments):
+            if isinstance(arg, KWArg):
+                if not hasattr(arg, 'cli_args'):
+                    cli_args = ('-{}'.format(arg.name), )
+                else:
+                    cli_args = arg.cli_args
+
+                if not hasattr(arg, 'cli_opts'):
+                    cli_opts = {}
+                else:
+                    cli_opts = arg.cli_opts
+
+                click.option(*cli_args, **cli_opts)(f)
+            elif isinstance(arg, Arg):
+                if not hasattr(arg, 'cli_args'):
+                    cli_args = (arg.name, )
+                else:
+                    cli_args = arg.cli_args
+
+                if not hasattr(arg, 'cli_opts'):
+                    cli_opts = {}
+                else:
+                    cli_opts = arg.cli_opts
+
+                click.argument(*cli_args, **cli_opts)(f)
+
+            else:
+                pass
+    return click.decorators.command()(f)
+
+
+def _iterate_tasks():
+    import_all_includes()
+    for extension in get_extensions():
+        if extension not in ('core', 'docker'):
+            for task in get_extension_tasks(extension).values():
+                yield task
```

Comment:
I get the convenience of just using the task names, but I do think it will be necessary to at least give the user's a signal about what plugin added the task.